includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -2291,13 +2291,14 @@ public function deserialize_intention_object_from_array( array $intention_array
 		$created = new DateTime();
 		$created->setTimestamp( $intention_array['created'] );
 
-		$charge_array       = 0 < $intention_array['charges']['total_count'] ? end( $intention_array['charges']['data'] ) : null;
-		$next_action        = ! empty( $intention_array['next_action'] ) ? $intention_array['next_action'] : [];
-		$last_payment_error = ! empty( $intention_array['last_payment_error'] ) ? $intention_array['last_payment_error'] : [];
-		$metadata           = ! empty( $intention_array['metadata'] ) ? $intention_array['metadata'] : [];
-		$customer           = $intention_array['customer'] ?? $charge_array['customer'] ?? null;
-		$payment_method     = $intention_array['payment_method'] ?? $intention_array['source'] ?? null;
-		$processing         = $intention_array[ Payment_Intent_Status::PROCESSING ] ?? [];
+		$charge_array         = 0 < $intention_array['charges']['total_count'] ? end( $intention_array['charges']['data'] ) : null;
+		$next_action          = ! empty( $intention_array['next_action'] ) ? $intention_array['next_action'] : [];
+		$last_payment_error   = ! empty( $intention_array['last_payment_error'] ) ? $intention_array['last_payment_error'] : [];
+		$metadata             = ! empty( $intention_array['metadata'] ) ? $intention_array['metadata'] : [];
+		$customer             = $intention_array['customer'] ?? $charge_array['customer'] ?? null;
+		$payment_method       = $intention_array['payment_method'] ?? $intention_array['source'] ?? null;
+		$processing           = $intention_array[ Payment_Intent_Status::PROCESSING ] ?? [];
+		$payment_method_types = $intention_array['payment_method_types'] ?? $intention_array['payment_method_types'] ?? [];
```

Comment:
If my eyes are still good, I think the two are the same?

Reply:
Looks the same to me.

Reply:
You were right, it was the same. I just fixed this in https://github.com/Automattic/woocommerce-payments/pull/5940/commits/5cd4bf3f45ee89fc5b59ef014a748e7e65fa9dea